zerver/views/user_groups.py

```diff
@@ -179,15 +178,16 @@ def delete_user_group(
 
 
 @require_user_group_edit_permission
-@has_request_variables
+@typed_endpoint
 def update_user_group_backend(
     request: HttpRequest,
     user_profile: UserProfile,
-    user_group_id: int = REQ(json_validator=check_int, path_only=True),
-    delete: Sequence[int] = REQ(json_validator=check_list(check_int), default=[]),
-    add: Sequence[int] = REQ(json_validator=check_list(check_int), default=[]),
+    *,
+    user_group_id: PathOnly[Json[int]],
+    delete: Optional[Json[List[int]]] = None,
+    add: Optional[Json[List[int]]] = None,
```

Comment:
Why are you changing the defaults from empty lists to `None`? That seems undesirable for how the downstream code uses it.

Reply:
`ruff` complains of using mutable data structures (e.g. `list`) for argument defaults and suggests replacing it with `None`

I think the reason is that a mutable default argument is shared between different calls to the same function because it's created only one time when the function is created.

Reply:
Sure, but there's other ways to fix that than potentially changing the API, like passing a `default_factory`, I think? Perhaps @kennethnrk can suggest one of them for this place.

Reply:
ok, Thanks for this clarification

Reply:
Why the redifinition of  `add` to `[]` ? the downstream code handles `None` cases

Reply:
If the downstream code handles `None` correctly then it should be fine to not redefine.

Reply:
I'm gonna rework this to a nicer way to do it.